installer/modelsDLTK.py

```diff
@@ -39,14 +39,38 @@ def addModels(login: str, modelPath: str, update: bool):
 	logger(yellow("Warning: Uploading, please BE CAREFUL! This can be dangerous."),forceConsoleOutput=True)
 	logger(f'You are going to be connected to {login} to write in folder {SCIPION_SOFTWARE_EM}.',forceConsoleOutput=True)
 	if input("Continue? YES/no\n") != 'YES':
-		sys.exit()
+		return 1
```

Comment:
If we are returning a code (which is actually a very good idea to indicate if the command failed or not), we should:
- `return 0` at the end of the function when it went right
- Retrieve the retcode in the function `modeGetModels` in file `'./xmipp'` and run `exitXmipp()` with that retcode always.
- Now that we are at it, it would be a good idea to adapt the rest of the functions to do something similar

Reply:
But I would not mix the installator and this kind of methods. This method is just for developers

Reply:
What do you mean by mix?
This should not mess with the installation behaviour, as it is not going to write in the config file, neither send an API call. We are only returning codes to exit the script with one code or another, to indicate the success/failure of the executed command, so that external software, like scipion, seeing such returncode, can interpret the state and act accordingly.

We can do one of two things.
1. Remove the `return 1` and leave it as a `return`. This will mean that in such case, the function will exit without returing anything (or `None` to be precise), but the `./xmipp ...` command will ultimately exit with retcode 0, meaning success, even if that was not the case. This is a very simple approach that requires little to no effort, but not the most robust one.
2. Handling return codes of all functions, both installation-related and non-installation related. It requires significant development effort (especially to properly test wether all functions **always** exit with a retcode), but it will allow the `'./xmipp'` script to exit it's execution with a retcode, which can allow wrapping it's commands properly in other software (like scipion, or bash scripts), where the code can react based on the success/failure of the executed xmipp command.

I'd rather go with the second option, but it's up to you to choose what you see fit.

Reply:
Mix: to have returns code related to the installation and also about other methods as this.
I would like to have more granularity returns code, specially in the cmake command (the main of the installation) but is out of my knowledge and out of my timeavailability. 
Therefore: option 1
Thanksss

Reply:
The return codes cannot mix. There is exactly only one retcode received every time you run `./xmipp ...`, which will be the retcode of that specific command, installation-related or not.

Regarding the granularity in CMake errors, @oierlauzi and I discussed this a while ago. As far as I know, CMake does not have a built-in way of returning different error codes deppending on which dependency is missing/under the minimum required. We agreed that trying to do this distinction in Python would mean reimplementing/copying the behaviour of how CMake looks for specific binaries/libraries, which is a bit like reinventing the wheel, and completely defeats the purpose of delegating the dependency check behaviour to an external package. So we settled on sacrificing this granularity in exchange for the robustness CMake brings, together with the maintainability improvements (a heck of a lot less code to develop-test-maintain).

Reply:
Understood